### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [oneapi-src/oneTBB#79]
Would you like to add more error handling for return values from functions like the following?
- [pthread_mutex_init](http://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_mutex_init.html "Initialize a mutex.") ⇒ [critical_section_v4 constructor](https://github.com/01org/tbb/blob/d31fcc68980efbe912dc8cec4978fbb77fb3cf9f/include/tbb/critical_section.h#L51)
- [__itt_mutex_lock](https://github.com/01org/tbb/blob/d31fcc68980efbe912dc8cec4978fbb77fb3cf9f/src/tbb/tools_api/ittnotify_config.h#L281 "Lock a mutex.") ⇒ [fini_ittlib](https://github.com/01org/tbb/blob/d31fcc68980efbe912dc8cec4978fbb77fb3cf9f/src/tbb/tools_api/ittnotify_static.c#L869)

**Comment by alexey-katranov** [comment]
It makes sense. Thank you for the report.

**Comment by anton-potapov** [comment]
`critical_section` was [removed](https://software.intel.com/content/dam/develop/external/us/en/documents/tbbrevamp.pdf) in oneTBB

**Comment by elfring** [comment]
:thinking: How does the mentioned “wrapping” improve the reported software situation?

**Comment by isaevil** [comment]
You are right, sorry. I didn't notice that return value handling is still unchanged.
Notify @ekovanova. The return value is still unchecked for `__itt_mutex_lock` at https://github.com/intel/ittapi/blob/master/src/ittnotify/ittnotify_static.c#L139

**Comment by arunparkugan** [comment]
@isaevil is this issue still relevant?

**Comment by elfring** [comment]
How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2C_C.2B.2B "Tools for code analysis") also for this software? :thinking:
